macros/src/lib.rs

```diff
@@ -545,21 +550,134 @@ pub fn option_setters_2(
                 self.options().#name = Some(#value);
                 self
             }
+        });
+    }
+
+    // Build rustdoc information.
+    let doc_name = args.doc_name;
+    let mut doc_impl = impl_in.clone();
+    // Synthesize a fn entry for each extra listed so it'll get a rustdoc entry
+    if let Some((_, extra)) = args.extra {
+        for name in &extra.names {
+            doc_impl.items.push(parse_quote! {
+                pub fn #name(&self) {}
+            });
+        }
+    }
+
+    // All done.  Export the tokens for doc use as their own distinct (uncompiled) item.
+    quote! {
+        #impl_in
+
+        #[macro_magic::export_tokens_no_emit(#doc_name)]
+        #doc_impl
+    }
+    .into()
+}
+
+struct OptionSettersArgs {
+    source_text: (Ident, Token![=]), // source =
+    foreign_path: syn::Path,
+    name_text: (Token![,], Ident, Token![=]), // , doc_name =
+    doc_name: Ident,
+    extra: Option<(Token![,], OptionSettersArgsExtra)>,
+}
+
+#[derive(Debug)]
+struct OptionSettersArgsExtra {
+    extra_text: (Ident, Token![=]), // extra =
+    bracket: Bracket,
+    names: Punctuated<Ident, Token![,]>,
+}
+
+impl Parse for OptionSettersArgs {
```

Comment:
`Parse` + `ToTokens` + `ForeignPath` is required for `macro_magic` custom arguments, it's a bit of a song and dance.  At least it's invisible at the call site, though.